Userland/Libraries/LibGfx/Painter.h

```diff
@@ -162,6 +162,27 @@ class Painter {
 
     int scale() const { return state().scale; }
 
+    template<typename TGetPixelCallback>
+    void fill_pixels(Gfx::IntRect const& region, TGetPixelCallback callback, bool blend = false)
```

Comment:
This feels like yet another arbitrary escape hatch in `Gfx::Painter` to allow some outside thing to have good performance.

I think it's okay for now, but longer term I'd like to see a re-think of `Gfx::Painter` as a canvas API with some kind of "brush" concept where a brush concept could be e.g color/pattern/gradient. And then all this kind of stuff would be handled inside LibGfx instead of poking random holes in the abstraction for LibWeb (and others) to access.